app/models/katello/rpm.rb

```diff
@@ -215,6 +216,13 @@ def self.applicable_to_hosts(hosts)
         where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts).distinct
     end
 
+    # Return RPMs that are not installed on a host, but could be installed
+    # the word 'installable' has a different meaning here than elsewhere
+    def self.yum_installable_for_host(host)
```

Comment:
There is another method `Katello::Rpm.installable_for_hosts` which gets all upgradable packages for hosts.
Maybe we should normalize all these method names so it is more meaningful and clear to the users.

Reply:
yeah.. Naming things is hard.  see https://github.com/Katello/katello/pull/9872#discussion_r789647196

@jturel @jlsherrill thoughts?

Reply:
How about `available_for_hosts`? Just some thought.

Reply:
I am coming into this conversation out of left field, so please take what I am saying with a grain of salt.

I'd write the method this way:

```
def self.installable_for_host(host:, only_missing_packages: false)
  repos = host.content_facet.bound_repositories.pluck(:id)
  rpms = self.in_repositories(repos).order(:name) # why do we order by name here? if it's for the sake of the view then I feel it's in the wrong place
  rpms = rpms.where.not(name: host.installed_packages.pluck(:name)) if only_missing_packages
  rpms
end
```

Why? I think the method name becomes more consistent with how we use installable elsewhere, and the extra flavor of making it only return packages which are not installed on the host is controlled by the param. Not sure I like the word 'missing' ...

Reply:
> `why do we order by name here? if it's for the sake of the view then I feel it's in the wrong place`

It's for readability of the yum install command rendered by the template. Does that count as the view? I don't want to deal with trying to do that in the template and adding more methods to the jail, etc.

`Katello::Rpm` already has an [`installable_for_hosts`](https://github.com/jeremylenz/katello/blob/7961da661073d22c9ae8e3f320186777aa85aa8b/app/models/katello/concerns/pulp_database_unit.rb#L86) method (via pulp_database_unit.rb), but it gets _applicable_ RPMs which by definition are already installed, and then filters them to only enabled repos. I feel like `installable_for_host` as a name is too similar to that method since what it does is so different.